src/core/server/saved_objects/service/lib/search_dsl/references_filter.test.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { getReferencesFilter } from './references_filter';
+
+describe('getReferencesFilter', () => {
+  const nestedRefMustClauses = (nestedMustClauses: unknown[]) => ({
+    nested: {
+      path: 'references',
+      query: {
+        bool: {
+          must: nestedMustClauses,
```

Comment:
Only added unit tests because the feature is already FTR covered in https://github.com/elastic/kibana/blob/e9d6a072a8123239c9aef9fc7a6eb16d74f27e71/test/api_integration/apis/saved_objects/find.ts#L269-L276